src/main/java/de/tum/cit/aet/artemis/communication/repository/CustomPostRepositoryImpl.java

```diff
@@ -50,6 +52,8 @@ public Page<Long> findPostIdsWithSpecification(Specification<Post> specification
         query.setMaxResults(pageable.getPageSize());
 
         List<Long> postIds = query.getResultList();
+        // removes all duplicates from the answer posts
+        List<Long> uniquePostIds = new ArrayList<>(new LinkedHashSet<>(postIds));
```

Comment:
Why do we even retrieve a list and not directly a set here?

Reply:
I now tried to return the set as a list by using distinct, but this is not possible. The search specification always contains a order by clause which leads to an hibernate error (by not returning the order-property).